integration/spark/src/main/java/io/openlineage/spark/agent/lifecycle/plan/LogicalRelationVisitor.java

```diff
@@ -60,41 +54,22 @@ public LogicalRelationVisitor(SparkContext context, String jobNamespace) {
   @Override
   public boolean isDefinedAt(LogicalPlan x) {
     return x instanceof LogicalRelation
-        &&
-        // ignore DatasetSources since they're handled by the DatasetSourceVisitor
-        !(((LogicalRelation) x).relation() instanceof DatasetSource);
+        && (((LogicalRelation) x).relation() instanceof HadoopFsRelation
+            || ((LogicalRelation) x).relation() instanceof JDBCRelation
+            || ((LogicalRelation) x).catalogTable().isDefined());
   }
 
   @Override
   public List<OpenLineage.Dataset> apply(LogicalPlan x) {
-    OpenLineage ol = new OpenLineage(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI);
     LogicalRelation logRel = (LogicalRelation) x;
     if (logRel.relation() instanceof HadoopFsRelation) {
       return handleHadoopFsRelation((LogicalRelation) x);
     } else if (logRel.relation() instanceof JDBCRelation) {
       return handleJdbcRelation((LogicalRelation) x);
     } else if (logRel.catalogTable().isDefined()) {
       return handleCatalogTable(logRel);
-    } else {
-      // make a best attempt at capturing the dataset information
-      log.warn("Don't know how to extract dataset from unknown relation {}", logRel.relation());
-      return Collections.singletonList(
-          PlanUtils.getDataset(
-              getUnknownDatasetName(logRel), jobNamespace, getDatasetFacet(ol, logRel)));
     }
-  }
-
-  private OpenLineage.DatasetFacets getDatasetFacet(OpenLineage ol, LogicalRelation logRel) {
-    return ol.newDatasetFacetsBuilder()
-        .documentation(ol.newDocumentationDatasetFacet(logRel.simpleString()))
-        .schema(PlanUtils.schemaFacet(logRel.schema()))
-        .build();
-  }
-
-  private String getUnknownDatasetName(LogicalRelation logRel) {
-    return logRel.relation().getClass().getSimpleName()
-        + "_"
-        + logRel.relation().schema().catalogString();
+    return Collections.emptyList();
```

Comment:
It's a problem if we end up here. if isDefinedAt is true but we don't find anything we should collect this information somehow so that we can fix the logic.

Reply:
possibly a warnings facet? Or does it go in the unknown inputs?

Reply:
I'll change this to throw an `IllegalStateException`. `isDefined` requires the `LogicalPlan` matches one of three cases and those three cases are covered here. We should never get to this line.